coverage.py

```diff
@@ -395,26 +396,47 @@ class CoverageProvider(BaseCoverageProvider):
     Turns errors in processing into coverage records with failure flags.
     """
 
-    # Does this CoverageProvider get its data from a source that also
-    # provides licenses for books?
-    CAN_CREATE_LICENSE_POOLS = False
+    def __init__(self, service_name, output_source, collection=None,
+                 input_identifier_types=None, batch_size=100,
+                 cutoff_time=None, operation=None, input_identifiers=None
+    ):
```

Comment:
There's a lot of cleanup here and adding of docstrings because I changed the signature of these constructors. I had to do a lot of work to make sure arguments were still being properly passed in.